src/adapter/src/coord.rs

```diff
@@ -2273,51 +2281,46 @@ impl Coordinator {
         &self,
         dataflow: &DataflowDescription<Plan>,
         cluster_id: ComputeInstanceId,
-        refresh_schedule: &Option<RefreshSchedule>,
     ) -> Antichain<Timestamp> {
         // All inputs must be readable at the chosen `as_of`, so it must be at least the join of
         // the `since`s of all dependencies.
         let id_bundle = dataflow_import_id_bundle(dataflow, cluster_id);
         let min_as_of = self.least_valid_read(&id_bundle);
 
-        // For compute reconciliation to recognize that an existing dataflow can be reused, we want
-        // to advance the `as_of` as far as possible. If a storage collection for the MV already
-        // exists, we can advance to that collection's upper. This is the most we can advance the
-        // `as_of` without skipping times in the MV output.
         let sink_id = dataflow
             .sink_exports
             .keys()
             .exactly_one()
             .expect("MV dataflow must export a sink");
-        let write_frontier = self.storage_write_frontier(*sink_id);
+
+        // For compute reconciliation to recognize that an existing dataflow can be reused, we want
+        // to advance the `as_of` far enough that it is beyond the `as_of`s of all dataflows that
+        // might still be installed on replicas, but we don't want to advance it so far that would
+        // prevent the warmup of a REFRESH MV. Similarly to `bootstrap_index_as_of`, we can use
+        // `least_valid_write_step_back`, but for MVs there is an additional constraint: we should
+        // also make sure that if a storage collection for the MV already exists, then we don't
+        // advance the `as_of` past that collection's `upper`, because then we'd skip times in the
+        // MV output.
```

Comment:
The storage collections needs to always exist now, so you can remove the "if a storage collection for the MV already exists" qualification.

Reply:
Thx